src/register/register.ts

```diff
@@ -15,7 +15,7 @@ export enum RegisterMode {
 };
 
 export interface IRegisterContent {
-  text               : string | string[];
+  text               : string | string[] | RecordedState;
```

Comment:
I directly append each action to `RecordedState.runActions` so we don't necessarily need an array.